### ISSUE TITLE
Types not working

### ISSUE THREAD

**Issue by franckdsf** [ShaMan123/erase2d#39]
Hey, I recently discovered the '@erase2d/fabric' library and attempted to integrate it into my Next.js project. However, I am encountering an issue with TypeScript type recognition when using this library with Fabric 6.0. Here are the details of the problem:

Steps to Reproduce:

1. Install the library and its dependencies in a Next.js project:
```bash
npm install @erase2d/fabric fabric@6.0.0-rc4
```
2. Attempt to import and use '@erase2d/fabric' in a TypeScript file (useCanvasEraser.ts).

```typescript
// Example useCanvasEraser.ts file
import { EraserBrush } from '@erase2d/fabric'; 
// Attempt to use EraserBrush but TypeScript does not recognize its types
```

Notice that TypeScript does not recognize types/interfaces from '@erase2d/fabric'.

> Also, I'm trying to use it to erase a *freeDrawingBrush*, but I haven't found any tutorials on how to do that

**Comment by ShaMan123** [comment]
Provide a next.js demo in codesanbox
/packages/app is exactly that and show how to use

**Comment by franckdsf** [comment]
Hey, the link for the codesandbox is [here](https://codesandbox.io/p/devbox/erase2d-fabric-type-issue-6y8np3?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clxwyx10n0007356jp1woj7td%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clxwyx10m0002356j8jnjkrda%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clxwyx10m0004356jr9ad09fl%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clxwyx10m0006356jsszczgz6%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B40%252C60%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clxwyx10m0002356j8jnjkrda%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxwyx10m0001356jmyts2ple%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%257D%255D%252C%2522id%2522%253A%2522clxwyx10m0002356j8jnjkrda%2522%252C%2522activeTabId%2522%253A%2522clxwyx10m0001356jmyts2ple%2522%257D%252C%2522clxwyx10m0006356jsszczgz6%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxwyx10m0005356j6vjhozlg%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clxwyx10m0006356jsszczgz6%2522%252C%2522activeTabId%2522%253A%2522clxwyx10m0005356j6vjhozlg%2522%257D%252C%2522clxwyx10m0004356jr9ad09fl%2522%253A%257B%2522id%2522%253A%2522clxwyx10m0004356jr9ad09fl%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxwyx10m0003356jw0t6kbuf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522npm%2520install%2520%2540erase2d%252Ffabric%25401.1.3%2522%252C%2522id%2522%253A%2522clxwyy2hg0052356j4w54vdv7%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clxwyx10m0003356jw0t6kbuf%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

> Yeah I figured it out, thank you :) I found it a bit challenging to locate, here is the [link](https://github.com/ShaMan123/erase2d/blob/0e9f7fcef8c9dea8a26ba0700815aedb4a03b508/packages/app/pages/index.tsx) for others who might have difficulty finding it

Also, I am trying to implement an undo method. I have successfully managed to undo the painted paths, but it seems that the eraser does not create new paths. Is there an event that I should listen to in order to track the erased paths?

```tsx
import type { Canvas, FabricObject } from "fabric";
import { useCallback, useEffect, useRef, useState } from "react";

export const useUndo = (canvas: Canvas | null) => {
  const undoListenerRef = useRef<VoidFunction>();
  const [paths, setPaths] = useState<Array<FabricObject>>([]);

  useEffect(() => {
    undoListenerRef.current = canvas?.on('path:created', ({ path }) => { setPaths((prev) => [...prev, path]) });
    return () => { undoListenerRef.current?.(); }
  }, [canvas])

  const undo = useCallback(() => {
    const path = paths.pop();
    if (path) { canvas?.remove(path); }
  }, [paths, canvas]);

  return { undo };
}
```

**Comment by ShaMan123** [comment]
now that types work again you should find examples by hovering `EraserBrush`, listen to the end event and manually commit erasing to get the data you need, I've added it to the example app, bump to 1.1.5.

**Comment by franckdsf** [comment]
Thanks for the quick update! I also noticed that you fixed the bug I had during canvas resize where the "clipPath" effect didn't scale properly (nice one, fixed in 1.1.6) 🎉

I've been playing around with your example, but I'm having a bit of trouble figuring out how to implement the undo method. I see that you've provided a way to "undo" using the ``invert`` property, but I'm more interested in a solution that can be linked to a *CTRL+Z* event.

In my current setup *(the hook above)*, I'm saving new paths and can successfully delete these newly created paths using the ``undo`` method. However, the same approach doesn't seem to work with the ``EraserBrush``. It looks like the commit function returns a map of objects, and deleting these objects results in some unwanted effects *(like the entire object getting deleted 😬)*.

So, I was wondering if the method I'm trying to implement is even feasible? 

Thanks in advance for your help!

**Comment by ShaMan123** [comment]
It is feasible.
The map returned is whst you should use.
You can access the values .parent property in order to remove the desired path.
Remember to mark both the object and its clipping group (.parent prop) as dirty before re-rendering to invalidate caches.
